src/sentry/platform_example/notification.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from django.utils import timezone
+
+from sentry import models
+from sentry.db.models.base import Model
+from sentry.platform_example.notification_digest_strategy import (
+    NotificationDigestBackend,
+    NotificationDigestStrategy,
+)
+from sentry.platform_example.notification_target import (
+    NotificationTarget,
+    NotificationTargetStrategy,
+    NotificationType,
+)
+
+
+class Notification(Model):
+    thread_id = models.CharField(null=False, max_length=128)
+    source = models.CharField(null=False, choices=NotificationType.get_choices(), max_length=30)
+    # Duplicated data from NotificationTarget, storing here for audit purposes
```

Comment:
could we just use a FK to maintain that relationship rather than duplicating the data?

Reply:
Notification Target is purely an in-memory concept with no DB backing in this initial design, so no FK would be possible for now 😅 I felt that maintaining a DB representation of these would be a ton of additional work, when we don't currently have a concept like this that maps to our current concepts of recipients/integrations. This is something we should be able to internalize/formalize in the future if we find it necessary though.

Reply:
If this is an in-memory only structure, you could use `dataclasses` instead of a django model.